.github/workflows/integration.yml

```diff
@@ -0,0 +1,213 @@
+# ------------------------------------------------------------
+# Copyright (c) Microsoft Corporation and Dapr Contributors.
+# Licensed under the MIT License.
+# ------------------------------------------------------------
+
+name: Components Integration Tests
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '*/30 * * * *'
+  pull_request:
+    branches:
+      - master
+      - release-*
+
+jobs:
+  pre_job:
+    name: Skip Duplicate Actions
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v3.4.0
+        with:
+          cancel_others: 'true'
+          paths_ignore: '["**.md", ".codecov.yaml", ".github/workflows/dapr-automerge.yml"]'
+  # Based on whether this is a PR or a scheduled run, we will run a different
+  # subset of the integration tests. This allows all the tests not requiring
+  # secrets to be executed on pull requests.
+  generate-matrix:
+    runs-on: ubuntu-latest
+    needs: pre_job
+    if: needs.pre_job.outputs.should_skip != 'true'
+    steps:
+    - name: Install yq
+      run: |
+        sudo snap install yq
+
+    - name: Specify components not requiring secrets nor certs
+      id: pr-components
+      run: |
+        PR_COMPONENTS=$(yq -I0 --tojson eval - << EOF
+        - bindings.mysql
+        - bindings.rabbitmq
+        - state.mysql
+        - state.postgresql
+        - state.sqlserver
+        EOF
+        )
+        echo "::set-output name=pr-components::$PR_COMPONENTS"
+
+    - name: Specify components requiring secrets or certs
+      id: cron-components
```

Comment:
Is there a reason we name this cron-components? From what I understand it seems like it runs at all times regardless and this just has to do with secrets. Is that correct?

Reply:
This naming is copied verbatim from conformance.yml: my understanding is that it's trying to distinguish which components run when triggered by PR (`pr-components`) and which ones only run on the cron trigger (`cron-components` i.e. any test that uses the pre-provisioned Azure test resource pool triggered by line 11)